tests/lockable_test.py

```diff
@@ -0,0 +1,80 @@
+"""Test locakble decorator."""
+from honeybee._lockable import lockable
+
+import pytest
+
+
+def test_simple_class():
+    """Test the application of the lockable decorator to a simple class."""
+    @lockable
+    class Foo(object):
+        def __init__(self):
+            self.bar = 10
+
+    foo = Foo()
+    foo.bar = 20
+
+    foo.lock()
+    with pytest.raises(AttributeError):
+        foo.bar = 30
+
+    foo.unlock()
+    foo.bar = 30
+    foo.lock()
+
+
+def test_nested_classes():
+    """Test lockable decorator with nested classes, which are not locked by default."""
+    @lockable
+    class Material(object):
+        def __init__(self):
+            self.r_value = 3
+
+    @lockable
+    class Construction(object):
+        def __init__(self):
+            self.material = Material()
+
+    constr = Construction()
+    constr.lock()
+    with pytest.raises(AttributeError):
+        constr.material = Material()
+
+    constr.material.r_value = 5
+    constr.material.lock()
```

Comment:
Interesting! Now I can see how overwriting `lock` and `unlock` can be helpful. In a construction for example you want to lock all the materials as well.